swagger.yaml

```diff
@@ -0,0 +1,515 @@
+      operationId: removeUserFromBillingAccount
+      responses:
+        '403':
+          description: No permission or wrong token
+          schema:
+            $ref: "#/definitions/ErrorModel"
+        '404':
+          description: Not found
+          schema:
+            $ref: "#/definitions/ErrorModel"
+        '204':
+          description: successful operation
+      parameters:
+        - $ref: "#/parameters/billingAcountIdParam"
+        - $ref: "#/parameters/userIdParam"
+
+  /me/billingaccounts:
+    get:
+      tags:
+        - billingaccount
+      security:
+        - Bearer: []
+      operationId: findBillingAccountsByUser
+      description: Retreive billing accounts that match the filter, For admin and non-admin user.
+      responses:
+        '403':
+          description: No permission or wrong token
+          schema:
+            $ref: "#/definitions/ErrorModel"
+        '422':
+          description: Invalid input
+          schema:
+            $ref: "#/definitions/ErrorModel"
+        '200':
+          description: A list of billing accounts
+          schema:
+            $ref: "#/definitions/BillingAccountListResponse"
+      parameters:
+        - $ref: "#/parameters/offsetParam"
+        - $ref: "#/parameters/limitParam"
+        - name: filter
+          required: true
+          type: string
+          in: query
+          description: |
+            Url encoded list of Supported filters
+             - status
+             - startDate
+             - endDate
+        - name: sort
+          required: false
+          description: |
+            sort billing accounts by status, startDate, endDate
+          in: query
+          type: string
+
+parameters:
+  billingAcountIdParam:
+    name: billingAccountId
+    in: path
+    description: billing account identifier
+    required: true
+    type: integer
+    format: int64
+  userIdParam:
+    name: userId
+    in: path
+    description: user identifier
+    required: true
+    type: integer
+    format: int64
+  offsetParam:
+    name: offset
+    description: "number of items to skip. Defaults to 0"
+    in: query
+    required: false
+    type: integer
+    format: int32
+  limitParam:
+    name: limit
+    description: "max records to return. Defaults to 10"
+    in: query
+    required: false
+    type: integer
+    format: int32
+
+definitions:
+  ResponseMetadata:
+    title: Metadata object for a response
+    type: object
+    properties:
+      totalCount:
+        type: integer
+        format: int64
+        description: Total count of the objects
+
+  ErrorModel:
+    type: object
+    properties:
+      id:
+        type: string
+        description: unique id identifying the request
+      version:
+        type: string
+      result:
+        type: object
+        properties:
+          success:
+            type: boolean
+          status:
+            description: http status code
```

Comment:
Wrapping responses is not uncommon, though it does seem redundant in most cases. For eg. we might want to return a different error code than http status to show that the server successfully processed the request but there were biz conditions that weren't met. Maybe there are http error codes that are appropriate for this specific use case though. The only other field I find useful is the response id, used to track logs across our system. But this can very well be moved to a response header.
I am all for simplifying the interface. Maybe we should review our API V3 spec.